### PR TITLE
Update Function interface

### PR DESCRIPTION
**Pull request by jq**
@deerzq @yolken @hectorgon @spencerde @christhetree

### CODE REVIEW

File: core/src/main/java/com/airbnb/aerosolve/core/function/Function.java [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
-
   // TODO rename numBins to something else, since it's a Spline specific thing
   Function aggregate(Iterable<Function> functions, float scale, int numBins);
 
-  float evaluate(float x);
+  float evaluate(float ... x);
 
-  void update(float x, float delta);
+  // delta is last element.
+  void update(float ... values);
```
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/function/Function.java]: how about `update(float delta, float ... values);` ?

Comment (jq) [inline core/src/main/java/com/airbnb/aerosolve/core/function/Function.java]: nice!

**Comment by jq** [comment]
@deerzq thanks for review, all updated!

File: core/src/test/java/com/airbnb/aerosolve/core/function/MultiDimensionPointTest.java [hunk 2]
```diff
@@ -64,46 +65,51 @@ public void getPointFromNDTree() throws Exception {
...
+    float[] af = new float[]{(float)1.0, (float)-2.0, (float)3.0, (float)4.0};
+    float[] bf = new float[]{(float)-5.0, (float)-6.0, (float)7.0, (float)8.0};
+    List<Float> a = Arrays.asList((float)1.0, (float)-2.0, (float)3.0, (float)4.0);
+
+    List<Float> b = Arrays.asList((float)-5.0, (float)-6.0, (float)7.0, (float)8.0);
+
     final double expected = Math.sqrt(84);
-    assertEquals(expected, MultiDimensionPoint.euclideanDistance(a, b), 0d);
-    assertEquals(expected, MultiDimensionPoint.euclideanDistance(b, a), 0d);
+    assertEquals(expected, MultiDimensionPoint.euclideanDistance(af, b), 0.001d);
```
Comment (deerzq) [inline core/src/test/java/com/airbnb/aerosolve/core/function/MultiDimensionPointTest.java]: `0.001f`

File: core/src/test/java/com/airbnb/aerosolve/core/function/MultiDimensionPointTest.java [hunk 3]
```diff
@@ -28,24 +29,24 @@ public void testEquals() throws Exception {
...
-        Arrays.asList(13.0, 15.0, 10.0)
+    List<List<Float>> r = Arrays.asList(
+        Arrays.asList((float)3.0, (float)-5.0, (float)-9.0),
+        Arrays.asList((float)13.0, (float)-5.0, (float)-9.0),
+        Arrays.asList((float)3.0, (float)15.0, (float)-9.0),
+        Arrays.asList((float)13.0, (float)15.0, (float)-9.0),
+        Arrays.asList((float)3.0, (float)-5.0, (float)10.0),
+        Arrays.asList((float)13.0, (float)-5.0, (float)10.0),
+        Arrays.asList((float)3.0, (float)15.0, (float)10.0),
+        Arrays.asList((float)13.0, (float)15.0, (float)10.0)
```
Comment (deerzq) [inline core/src/test/java/com/airbnb/aerosolve/core/function/MultiDimensionPointTest.java]: you can use `13.0f, 15.0f, 10.0f`

**Comment by deerzq** [comment]
some nits, overall LGTM

**Pull request merged**